boards/samr21-xpro/include/periph_conf.h

```diff
@@ -120,6 +120,101 @@ extern "C" {
+#define ADC_0_RUN_IN_STANDBY               0 /* disabled */
+
+/* ADC 0 Module Status flags */
+#define ADC_0_STATUS_RESULT_READY          (1UL << 0)
+#define ADC_0_STATUS_WINDOW                (1UL << 1)
+#define ADC_0_STATUS_OVERRUN               (1UL << 2)
+
+/* ADC 0 Positive Input Pins */
+#define ADC_0_POS_INPUT                    ADC_INPUTCTRL_MUXPOS_PIN6
+
+/* ADC 0 Negative Input Pins */     
+#define ADC_0_NEG_INPUT                    ADC_INPUTCTRL_MUXNEG_GND
+
+/* ADC 0 Gain Factor */
+#define ADC_0_GAIN_FACTOR_1X               ADC_INPUTCTRL_GAIN_1X
+#define ADC_0_GAIN_FACTOR_2X               ADC_INPUTCTRL_GAIN_2X
+#define ADC_0_GAIN_FACTOR_4X               ADC_INPUTCTRL_GAIN_4X
+#define ADC_0_GAIN_FACTOR_8X               ADC_INPUTCTRL_GAIN_8X
+#define ADC_0_GAIN_FACTOR_16X              ADC_INPUTCTRL_GAIN_16X
+/* Use this to define the value used */
+#define ADC_0_GAIN_FACTOR_DEFAULT          ADC_0_GAIN_FACTOR_1X
+
+/* ADC 0 Resolutions */    
+#define ADC_0_RES_8BIT                     ADC_CTRLB_RESSEL_8BIT
+#define ADC_0_RES_10BIT                    ADC_CTRLB_RESSEL_10BIT
+#define ADC_0_RES_12BIT                    ADC_CTRLB_RESSEL_12BIT
+#define ADC_0_RES_16BIT                    ADC_CTRLB_RESSEL_16BIT 
+
+/* ADC 0 Voltage reference */
+#define ADC_0_REF_INT_1V                   ADC_REFCTRL_REFSEL_INT1V
+#define ADC_0_REF_EXT_B                    ADC_REFCTRL_REFSEL_AREFB
+#define ADC_0_REF_COM_EN                   0
+/* Use this to define the value used */ 
+#define ADC_0_REF_DEFAULT                  ADC_0_REF_INT_1V
+
+/* ADC 0 ACCUMULATE */
+#define ADC_0_ACCUM_DISABLE                ADC_AVGCTRL_SAMPLENUM_1
+#define ADC_0_ACCUM_2                      ADC_AVGCTRL_SAMPLENUM_2
+#define ADC_0_ACCUM_4                      ADC_AVGCTRL_SAMPLENUM_4
+#define ADC_0_ACCUM_8                      ADC_AVGCTRL_SAMPLENUM_8
+#define ADC_0_ACCUM_16                     ADC_AVGCTRL_SAMPLENUM_16
+#define ADC_0_ACCUM_32                     ADC_AVGCTRL_SAMPLENUM_32
+#define ADC_0_ACCUM_64                     ADC_AVGCTRL_SAMPLENUM_64
+#define ADC_0_ACCUM_128                    ADC_AVGCTRL_SAMPLENUM_128
+#define ADC_0_ACCUM_256                    ADC_AVGCTRL_SAMPLENUM_256
+#define ADC_0_ACCUM_512                    ADC_AVGCTRL_SAMPLENUM_512
+#define ADC_0_ACCUM_1024                   ADC_AVGCTRL_SAMPLENUM_1024
+/* Use this to define the value used */
+#define ADC_0_ACCUM_DEFAULT                ADC_0_ACCUM_DISABLE 
+
+/* ADC 0 DEVIDE RESULT */
+#define ADC_0_DIV_RES_DISABLE              0
+#define ADC_0_DIV_RES_2                    1
+#define ADC_0_DIV_RES_4                    2
+#define ADC_0_DIV_RES_8                    3
+#define ADC_0_DIV_RES_16                   4
+#define ADC_0_DIV_RES_32                   5
+#define ADC_0_DIV_RES_64                   6
+#define ADC_0_DIV_RES_128                  7
+/* Use this to define the value used */
+#define ADC_0_DIV_RES_DEFAULT             ADC_0_DIV_RES_DISABLE 
+/** @} */
+
```

Comment:
This could be done for all peripherals as well, BSS is very precious on the SAMR21.
I don't have any device to try this PR unfortunately.
@wiredsource, Could you give us a TODO list with what is missing for merging this PR and/or the issues please.